### PR TITLE
impl new derive macro to replace attribute macro

### PR DESCRIPTION
**Pull request by Kilerd**
this PR is to impl issue #58 

main change: 
- remove `#[tonbo_record]` macro, use `#[derive(Record)]` instead
- remove `#[primary_key]` attribute, use `#[record(primary_key)]`
- remove `#[tonbo_record(::serde::Serialize)]` logic, use `#[derive(Record, ::serde::Serialize)]` instead
- remove builtin trait derive impl `::tonbo::KeyAttributes, Debug, PartialEq, Eq, Clone`

refactor related:
- separate record code gen into small handler for human-friendly maintenance
- extract ident generator util for Centralized Management 

further optimization:
- detect multiple primary_key definition (had added todo item in codebase)
- rename generated struct's name for code isolation, like we will generate struct `UserRef` for struct `User`, which may cause code conflict if use had defined one in his codebase. the common pattern is to add unique prefix for generated struct , E.G. `__TonboUserRef`
- possible to remove trait `Debug` as required trait for trait `Record`, cuz it will slow down the compile time. and have no idea why we need this trait.

### CODE REVIEW

**Comment by Kilerd** [comment]
and BTW I dont use tonbo in any of my projects, if you guys can provide some testcases are very helpful.

**Comment by KKould** [comment]
> and BTW I dont use tonbo in any of my projects, if you guys can provide some testcases are very helpful.

This is a very big and meaningful PR, I will continue to follow up on the testing work

**Comment by Kilerd** [comment]
yep, you're right, and I recommend to review this PR commit by commit.

**Comment by KKould** [comment]
@Kilerd @ethe at present, macros does not support many configurable functions. Its core is to help users reduce code, so I only add tests to macro to verify the correctness. Please take a look

File: src/lib.rs [hunk 1]
```diff
@@ -852,8 +852,8 @@ pub(crate) mod tests {
 
         fn size(&self) -> usize {
             let string_size = self.vstring.len();
-            let u32_size = std::mem::size_of::<u32>();
-            let bool_size = self.vbool.map_or(0, |_| std::mem::size_of::<bool>());
+            let u32_size = mem::size_of::<u32>();
```
Comment (Kilerd) [inline src/lib.rs:855]: why to remove the `std` prefix

Comment (KKould) [reply src/lib.rs:855]: Because this is not implemented in macro, this path can be removed (shown in gray in rustrover)

Comment (Kilerd) [reply src/lib.rs:855]: make sense

**Review by ethe — APPROVED** [review]

**Comment by ethe** [comment]
This is pretty neat! Thanks to @Kilerd

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -45,6 +45,7 @@ jobs:
         uses: actions-rs/cargo@v1
         with:
           command: test
+          args: --all
```
Comment (ethe) [inline .github/workflows/ci.yml:48]: `--workspace` instead?

**Review by KKould — APPROVED** [review]
LGTM

**Pull request merged**